be/src/storage/lake/lake_primary_index.cpp

```diff
@@ -113,53 +114,81 @@ Status LakePrimaryIndex::_do_lake_load(Tablet* tablet, const TabletMetadata& met
             CHECK(false) << "create column for primary key encoder failed";
         }
     }
-    vector<uint32_t> rowids;
-    rowids.reserve(4096);
-    auto chunk_shared_ptr = ChunkHelper::new_chunk(pkey_schema, 4096);
-    auto chunk = chunk_shared_ptr.get();
     // 2. scan all rowsets and segments to build primary index
     auto rowsets = tablet->get_rowsets(metadata);
     if (!rowsets.ok()) {
         return rowsets.status();
     }
     // NOTICE: primary index will be builded by segment files in metadata, and delvecs.
     // The delvecs we need are stored in delvec file by base_version and current MetaFileBuilder's cache.
-    for (auto& rowset : *rowsets) {
-        auto res = rowset->get_each_segment_iterator_with_delvec(pkey_schema, base_version, builder, &stats);
-        if (!res.ok()) {
-            return res.status();
+    if (config::use_lake_pk_index_loader) {
```

Comment:
the configuration is runtime changable, what if it set to false during the startup and turned on during runtime, the lake_pk_index_loader() will be a nullptr, right?

Reply:
lake_pk_index_loader will be always initialized in shard_data mode.